src/main/java/rx/internal/operators/OnSubscribeRedo.java

```diff
@@ -235,8 +235,9 @@ public void onError(Throwable e) {
                     @Override
                     public void onNext(T v) {
                         if (!done) {
-                            if (consumerCapacity.get() != Long.MAX_VALUE) {
-                                consumerCapacity.decrementAndGet();
+                            long cap = consumerCapacity.get();
```

Comment:
Unfortunately still fails, now with less frequency in the 5000 attempts as before.

I'll adjust the PR anyway to reflect what you suggest.